test/promql/tests/data_plane/gitops_routes_test.yaml

```diff
@@ -0,0 +1,408 @@
+evaluation_interval: 1m
+
+rule_files:
+  - prometheus.gitops_routes_alerts.yaml
+
+tests:
+  - interval: 1m
+    input_series:
+      - series: 'openshift_route_status{status="True", name="route-001", namespace="gitops-service-argocd"}'
+        values: '0 0 0 0 0 0 0 0 0 0 0'
+
+      - series: 'openshift_route_status{status="False",  name="route-002", namespace="gitops-service-argocd"}'
+        values: '1 1 1 1 1 1 1 1 1 1 1'
```

Comment:
Does the `openshift_route_status` metric gives `0` on the non-current status, or does it give no data?
e.g. when `openshift_route_status{status="False"}` is at `1`, would `openshift_route_status{status="True"}` would be on `0` or it will just not provide any data?

Reply:
If `openshift_route_status{status="False"}` is at 1 `openshift_route_status{status="True"}` will have no row.

Reply:
If I understand correctly, that metric cannot have a value of zero. If that is the case, I suggest dropping the test cases having that value, as they might be misleading for the behavior of the metric.

Reply:
The test case that I think we're missing is the one in which we have a single route failing. It should be something like that:
```
      - series: 'openshift_route_status{status="True", name="route-001", namespace="gitops-service-argocd"}'
        values: '_ _ _ _ _ _ _ _ _ _ _ '

      - series: 'openshift_route_status{status="False",  name="route-001", namespace="gitops-service-argocd"}'
        values: '1 1 1 1 1 1 1 1 1 1 1'
```
I think that one might not be working, as the label-true query will not return any time series and no calculation will be performed (and no alert).

This sort of metric is tricky to do calculations on. We had long discussions on a metric with similar behavior here: https://github.com/redhat-appstudio/o11y/pull/94

Have a look to see if it helps you.

Reply:
Yes initially I tried giving value `"_ _ _ _ _"` but as you said it fails so I used 0, but it seems that changes the scenario.